### PR TITLE
Pagina dashboard

### PR DESCRIPTION
**Pull request by CamiloArias12**
# Page dashboard

## Description
Creacion pagina principal responsive
### Summary
image
image

## How Has This Been Tested?
Se testeo directamente en la web

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/modules/auth/services/actions.ts [hunk 1]
```diff
@@ -22,3 +33,72 @@ export async function authenticate(
...
+    })
+
+    if (validateFields.success) {
+        if (validateFields.data.password === validateFields.data.passwordReply) {
+            const response = await changePasswordService(validateFields.data.token, validateFields.data.password)
+            if (response) {
+                return {
+                    message: 'La contrasena ha sido actualizada'
+                }
+
```
Comment (Alejandro-Suarez-Rincon) [inline src/modules/auth/services/actions.ts:85]: Dar formato

```suggestion

```

File: src/modules/auth/services/actions.ts [hunk 2]
```diff
@@ -22,3 +33,72 @@ export async function authenticate(
...
+            } else {
+                return {
+                    errors: {
+                    },
+                    message: 'Error'
+                }
+
+
```
Comment (Alejandro-Suarez-Rincon) [inline src/modules/auth/services/actions.ts:93]:
```suggestion
```

File: src/modules/auth/services/actions.ts [hunk 3]
```diff
@@ -22,3 +33,72 @@ export async function authenticate(
...
+            }
+
+
```
Comment (Alejandro-Suarez-Rincon) [inline src/modules/auth/services/actions.ts:96]:
```suggestion
```

**Review by Alejandro-Suarez-Rincon — APPROVED** [review]

File: src/app/change-password/page.tsx [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { ErrorValidatePasswordReset } from "@/modules/auth/components/error-validate-password-reset";
+import { validateTokenPasswordResetService } from "@/modules/auth/services";
+import ChangePassword from "@/modules/auth/templates/change-password";
+
+export const metadata = {
+    title: "OptiTech",
+};
+
+export default async function Page({ searchParams }: { searchParams: { token: string } }) {
+
```
Comment (Camilo-Camargo) [inline src/app/change-password/page.tsx:10]:
```suggestion
```

File: src/auth.config.ts [hunk 5]
```diff
@@ -19,26 +22,26 @@ export const authConfig: NextAuthConfig = {
...
-                return NextResponse.redirect(new URL('/', nextUrl));
-            }
 
-            if (!isLoggedIn) {
-                return NextResponse.redirect(new URL(ROUTES_AUTH.PRINCIPAL, nextUrl));
+            if (isLoggedIn && publicRoutes.has(pathname)) {
+                return NextResponse.redirect(new URL(ROUTES_AUTH.DASHBOARD, nextUrl));
             }
-
+
```
Comment (Camilo-Camargo) [inline src/auth.config.ts:40]:
```suggestion
```

File: src/auth.ts [hunk 6]
```diff
@@ -13,16 +14,32 @@ export const { auth, handlers, signIn, signOut } = NextAuth({
                     .safeParse(credentials);
                 if (parsedCredentials.success) {
                     const { email, password } = parsedCredentials.data;
-                    if (email === "optitech@optitech.com" && password === "optitech")
-                        return {
-                            name: "Optitech",
-                            email: "optitech@gmail.com"
-                        }
+                    console.log(email, password)
```
Comment (Camilo-Camargo) [inline src/auth.ts:17]: eliminar esto

File: src/modules/auth/components/error-validate-password-reset/index.tsx [hunk 7]
```diff
@@ -0,0 +1,17 @@
+"use client"
+import { useRouter } from "next/navigation"
+import { useEffect } from "react"
+import { toast } from "react-toastify"
+
+export function ErrorValidatePasswordReset() {
+    useEffect(() => {
+
```
Comment (Camilo-Camargo) [inline src/modules/auth/components/error-validate-password-reset/index.tsx:8]:
```suggestion
```

File: src/modules/auth/services/actions.ts [hunk 8]
```diff
@@ -22,3 +33,67 @@ export async function authenticate(
...
+            }
+        }
+    }
+    return {
+        errors: validateFields.error.flatten().fieldErrors,
+        message: 'Error'
+    }
+
+}
+
```
Comment (Camilo-Camargo) [inline src/modules/auth/services/actions.ts:66]: invertir if

File: src/modules/auth/services/actions.ts [hunk 9]
```diff
@@ -22,3 +33,67 @@ export async function authenticate(
...
+                }
+            }
+        }
+    }
+    return {
+        errors: validateFields.error?.flatten().fieldErrors,
+        message: 'Error'
+    }
+
+}
```
Comment (Camilo-Camargo) [inline src/modules/auth/services/actions.ts:99]: invertir if

File: src/modules/auth/services/index.ts [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+            },
+            body: JSON.stringify({ email, password })
+        });
+        if (response.ok) {
+            const data = await response.json();
+            return data.token
+        } else {
+            console.error('Login failed:', response.statusText);
+            return null;
+        }
```
Comment (Camilo-Camargo) [inline src/modules/auth/services/index.ts:34]: invertir if

File: src/modules/auth/services/index.ts [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+export async function changePasswordService(token: string, password: string): Promise<boolean> {
+    try {
+        const response = await fetch(`${process.env.API_URL}/client/reset-password-token`,
+            {
+                method: 'POST',
+                headers: {
+                    "Content-Type": "application/json",
+                },
+                body: JSON.stringify({ token: token, password: password })
+
```
Comment (Camilo-Camargo) [inline src/modules/auth/services/index.ts:50]: utilizar api post del anterior pr https://github.com/OwlByTech/optitech-web/pull/6#discussion_r1651978380

File: src/modules/auth/services/index.ts [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+            })
+        return await response.json()
+    } catch (e) {
+        return false
+    }
+}
+
```
Comment (Camilo-Camargo) [inline src/modules/auth/services/index.ts:57]:
```suggestion
```

File: src/modules/auth/types/index.ts [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+export enum ROUTES_AUTH {
+    LOGIN = "/login",
+    SING_UP = "/sign-up",
+    DASHBOARD = "/dashboard",
+    RESET_PASSWORD = "/reset-password",
+    CHANGE_PASSWORD = "/change-password"
+}
+
```
Comment (Camilo-Camargo) [inline src/modules/auth/types/index.ts:13]:
```suggestion
```

File: src/modules/dashboard/components/profile/index.tsx [hunk 14]
```diff
@@ -0,0 +1,19 @@
+import { CgProfile } from "react-icons/cg";
+import { clientInfoService } from "../..";
+
+export async function Profile() {
+
```
Comment (Camilo-Camargo) [inline src/modules/dashboard/components/profile/index.tsx:5]:
```suggestion
```

File: src/modules/dashboard/components/profile/index.tsx [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+    return <div className=" flex flex-row gap-2 ">
+        <img className="bg-white md:bg-gray-200 text-xs  text-black h-12 w-12 " src="/profile.png" />
+        <div className=" flex flex-col justify-end text-xs">
+            <span>
+                {`${clientInfo.givenName} ${clientInfo.surname}`}
+            </span>
+            <span>
+                {clientInfo.email}
+            </span>
+
```
Comment (Camilo-Camargo) [inline src/modules/dashboard/components/profile/index.tsx:16]:
```suggestion
```

File: src/modules/dashboard/components/routes-sidebar/index.tsx [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+import Link from "next/link";
+import { route } from "../../types";
+
+type Props = {
+    routes: route[]
+    path: string
+}
+
+export function RoutesSidebar({ routes, path }: Props) {
+
```
Comment (Camilo-Camargo) [inline src/modules/dashboard/components/routes-sidebar/index.tsx:11]:
```suggestion
```

File: src/modules/dashboard/index.ts [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+                "Authorization": `Bearer ${session?.user.token}`
+            }
+        });
+        if (response.ok) {
+            // TODO: add types to the client Info service
+            return await response.json();
+        } else {
+            console.error('Login failed:', response.statusText);
+            return null;
+        }
```
Comment (Camilo-Camargo) [inline src/modules/dashboard/index.ts:18]: invertir if

File: src/modules/dashboard/templates/routes.tsx [hunk 18]
```diff
@@ -0,0 +1,15 @@
+"use client"
+import { usePathname } from "next/navigation";
+import { RoutesSidebar } from "../components/routes-sidebar";
+import { routesSidebarDown, routesSidebarUp } from "../types";
+
+export function Routes() {
+
```
Comment (Camilo-Camargo) [inline src/modules/dashboard/templates/routes.tsx:7]:
```suggestion
```

File: src/modules/dashboard/templates/routes.tsx [hunk 19]
```diff
@@ -0,0 +1,15 @@
...
+    const path = usePathname()
+
```
Comment (Camilo-Camargo) [inline src/modules/dashboard/templates/routes.tsx:9]:
```suggestion
```

File: src/modules/dashboard/templates/sidebar.tsx [hunk 20]
```diff
@@ -0,0 +1,10 @@
+import { Profile } from "../components/profile";
+import { Routes } from "./routes";
+
+export function SideBar() {
+    return <div className=" flex h-full flex-col pt-12  pb-5  px-5 gap-4">
+        <Profile />
+        <Routes />
+
```
Comment (Camilo-Camargo) [inline src/modules/dashboard/templates/sidebar.tsx:8]:
```suggestion
```

File: src/modules/dashboard/types/index.tsx [hunk 21]
```diff
@@ -0,0 +1,50 @@
...
+    ASESOR = "/dashboard/asesor",
+    HELP = "/dashboard/help"
+}
+export type route = {
+    route: string
+    name: string
+    icon: ReactNode
+}
+
+
```
Comment (Camilo-Camargo) [inline src/modules/dashboard/types/index.tsx:17]:
```suggestion
```

**Review by JosephSC0121 — COMMENTED** [review]
Hay muchos cambios que ya están en otros prs

File: src/modules/dashboard/index.ts (hunk 17 above)
Comment (JosephSC0121) [reply src/modules/dashboard/index.ts:18]: https://medium.com/swlh/return-early-pattern-3d18a41bba8
https://refactoring.guru/es/replace-nested-conditional-with-guard-clauses

**Review by Alejandro-Suarez-Rincon — APPROVED** [review]

**Review by JosephSC0121 — APPROVED** [review]

File: src/auth.ts (hunk 6 above)
Comment (CamiloArias12) [reply src/auth.ts]: [d974c8c](https://github.com/OwlByTech/optitech-web/pull/7/commits/d974c8c15e395001e2965a28647661be28fa82eb)

**Pull request merged**